### DISCUSSION TITLE
External redirect from onBeforeRender

### DISCUSSION THREAD

**Discussion by oleghalin** [vikejs/vike#589] [Discussion] [answered]
Hello. How can i make external redirect from onBeforeRender hook? I tried to use pageContext.redirectTo but i see only errors on render stage.

**Comment by brillout** [comment, marked as answer]
https://vite-plugin-ssr.com/page-redirection#page-content
https://github.com/brillout/rules

**Reply by oleghalin** [reply]
Yeah, I figured it out, maybe we should add some bookmark that it also requires changes on hook side, I expected that it comes with vps plug-in by default
Thanks

**Reply by brillout** [reply]
Agreed & done: https://github.com/brillout/vite-plugin-ssr/commit/6b674fcae42c60d264446b27d9fd178a5e74341e.
